lib/puppet/provider/service/launchd.rb

```diff
@@ -196,6 +196,22 @@ def self.read_plist(path)
     Puppet::Util::Plist.read_plist_file(path)
   end
 
+  # Read overrides plist, retrying if necessary
+  def self.read_overrides
+    i = 1
+    overrides = nil
+    loop do
+      Puppet.debug("Reading overrides plist, attempt #{i}") if i > 1
+      overrides = read_plist(launchd_overrides)
+      break unless overrides.nil?
+      raise Puppet::Error.new('Unable to read overrides plist, too many attempts') if i == 20
+      Puppet.info('Overrides file could not be read, trying again.')
+      sleep(0.1)
```

Comment:
Would be nice to reference this as `Kernel.sleep` to make it explicit that that's what we're calling.

Reply:
Done.